frontend/src/pages/miscellaneous/ProfilePage.tsx

```diff
@@ -182,6 +192,51 @@ export default function ProfilePage() {
         window.location.reload();
     };
 
+    const handleImageChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const file = event.target.files?.[0];
+        const id = localStorage.getItem('id') || '';
+        console.log("file in react", file);
+        if (file) {
+            const formData = new FormData();
+            formData.append('image', file);
+            formData.append('userId', id);
+            axios.post('http://localhost:4001/users/upload', formData, {
+                headers: {
+                  'Content-Type': 'multipart/form-data',
+                },
+            })
+            .then(response => {
+                localStorage.setItem('profileImage', response.data.fileName)
+                getUserProfilePic(response.data.fileName)
+                console.log('Image uploaded successfully:', response.data.message);
+            })
+            .catch(error => {
+                console.error('Error uploading image:', error);
+            });
+        }
+      };
+
+      // TODO: Make it a GET request instead
+      const getUserProfilePic = async (imageName: string) => {
```

Comment:
Similarly, handling of axios should not be here, should be handled under src/services/user-service. This function should be moved to the services/user-service. Then you call that function here. Once you move to the services/user-service, you can call this function for the authenticated navbar, encouraging reusability.